src/controllers/payment.controller.js

```diff
@@ -0,0 +1,173 @@
+				cart: JSON.stringify(cart),
+				paymentId: payment.id,
+			},
+		});
+		const session = await stripe.checkout.sessions.create({
+			payment_method_types: ['card'],
+			shipping_address_collection: {
+				allowed_countries: ['RW', 'UG'],
+			},
+			shipping_options: [
+				{
+					shipping_rate_data: {
+						type: 'fixed_amount',
+						fixed_amount: {
+							amount: 0,
+							currency: 'rwf',
+						},
+						display_name: 'Free shipping',
+						delivery_estimate: {
+							minimum: {
+								unit: 'business_day',
+								value: 5,
+							},
+							maximum: {
+								unit: 'business_day',
+								value: 8,
+							},
+						},
+					},
+				},
+				{
+					shipping_rate_data: {
+						type: 'fixed_amount',
+						fixed_amount: {
+							amount: 10000,
+							currency: 'rwf',
+						},
+						display_name: 'Next day air',
+						delivery_estimate: {
+							minimum: {
+								unit: 'business_day',
+								value: 1,
+							},
+							maximum: {
+								unit: 'business_day',
+								value: 1,
+							},
+						},
+					},
+				},
+			],
+			phone_number_collection: {
+				enabled: true,
+			},
+			customer: customer.id,
+			line_items: lineItems,
+			mode: 'payment',
+			success_url: `${process.env.PAYMENT_BASE_URL}/success`,
+			cancel_url: `${process.env.PAYMENT_BASE_URL}/cancel`,
+		});
+		return res.status(200).json({
+			cartTotal: cartTotal,
+			message: 'Creating checkout session successful',
+			url: session.url,
+		});
+	} catch (error) {
+		return res
+			.status(500)
+			.json({
+				error: error.message,
+				message: 'Could not create checkout session',
+			});
+	}
+};
+
+export const webHook = async (req, res) => {
+  const sig = req.headers['stripe-signature'];
+  let data;
+  let event;
+  let eventType;
+  try {
+    event = stripe.webhooks.constructEvent(req.rawBody, sig, endpointSecret);
+    data = event.data.object;
+    eventType = event.type;
+  } catch (err) {
+    return res.status(400).json({
+      error: err.message,
+      message: 'Could not validate event signature',
+    });
+  }
+
+  let order;
+  if (eventType === 'checkout.session.completed') {
+    try {
+      const customer = await stripe.customers.retrieve(data.customer);
+      order = await OrderService.createNewOrder(customer, data);
+      const payment = await PaymentService.updatePayment({ status: 'Paid' }, order.paymentId);
+      const cart = JSON.parse(customer.metadata.cart);
+      const { products } = cart;
+      const orderId = order.id;
+      const sales = await Promise.all(
+        products.map(async (product, index) => {
+          const sale = {
+            orderId,
+            productId: product.productId,
+            quantitySold: products[index].quantity,
+          };
+          return await SalesService.createSales(sale);
+        })
+      );
+
+      return res.status(200).json({
+        message: 'Order created successfully',
+      });
+    } catch (error) {
+      return res.status(500).json({
```

Comment:
Let's refactor this controller by splitting into different methods and middleware if they are